From 786ea40882cf7e1fac3890bc0b845284d2ed5347 Mon Sep 17 00:00:00 2001
From: Andris PE <neandris@gmail.com>
Date: Sat, 8 Jun 2024 14:30:35 +0300
Subject: [PATCH] Offloading can also happen when ramips hw offload rejects on
 ICMP3

Additionally since jump target is terminal no need to preserve callaback
and use goto in place of jump.
---
 root/usr/share/firewall4/templates/ruleset.uc | 2 +-
 tests/01_configuration/01_ruleset             | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/root/usr/share/firewall4/templates/ruleset.uc
+++ b/root/usr/share/firewall4/templates/ruleset.uc
@@ -136,7 +136,7 @@ table inet fw4 {
 
 {% fw4.includes('chain-prepend', 'forward') %}
 {% if (length(flowtable_devices) > 0): %}
-		ct state vmap { established : jump handle_offload, related : accept } comment "!fw4: Handle forwarded flows"
+		ct state vmap { established : goto handle_offload, related : goto handle_offload } comment "!fw4: Handle forwarded flows"
 {% else %}
 		ct state vmap { established : accept, related : accept } comment "!fw4: Accept forwarded flows"
 {% endif %}
--- a/tests/01_configuration/01_ruleset
+++ b/tests/01_configuration/01_ruleset
@@ -122,7 +122,7 @@ table inet fw4 {
 	chain forward {
 		type filter hook forward priority filter; policy drop;
 
-		ct state vmap { established : jump handle_offload, related : accept } comment "!fw4: Handle forwarded flows"
+		ct state vmap { established : goto handle_offload, related : goto handle_offload } comment "!fw4: Handle forwarded flows"
 		iifname "br-lan" jump forward_lan comment "!fw4: Handle lan IPv4/IPv6 forward traffic"
 		iifname "pppoe-wan" jump forward_wan comment "!fw4: Handle wan IPv4/IPv6 forward traffic"
 		jump handle_reject
